src/charmed_kubeflow_chisme/testing/cos_integration.py

```diff
@@ -109,22 +102,30 @@ async def deploy_and_assert_grafana_agent(
     await model.wait_for_idle(apps=[GRAFANA_AGENT_APP], status="blocked", timeout=5 * 60)
 
 
-async def _check_metrics_endpoint(app: Application, metrics_endpoint: str) -> None:
-    """Check metrics endpoint accessibility.
+def _check_url(url: str, port: int, path: str) -> bool:
+    """Check port and path in url."""
```

Comment:
Nit

```suggestion
    """Return False if X, True otherwise.
    
        Check that the expected port and path are in the url after parsing it ...
        
        Args:
          url(str): something something
          ...
```

Reply:
I believe the `Args` bit is still missing

Reply:
I did not add `Args`, since all "private" functions are missing it. Do you think it's a good idea to add it to all? IMO, type hinting is enough.